Simplenote/StorageSettings.swift

```diff
@@ -8,29 +8,72 @@
 
 import Foundation
 
+enum StorageLocation {
+    case shared
+    case legacy
+}
+
 class StorageSettings {
-    let fileManager: FileManager
+    private let fileManager: FileManager
+    private(set) var storageLocation: StorageLocation
 
-    init(fileManger: FileManager = .default) {
+    init(fileManger: FileManager = .default, storageLocation: StorageLocation = .shared) {
         self.fileManager = fileManger
-    }
-
-    var userLibraryDirectory: URL? {
-        let libraryURL = fileManager.urls(for: .libraryDirectory, in: .userDomainMask).last
-        return libraryURL?.appendingPathComponent(Constants.modelName)
+        self.storageLocation = storageLocation
     }
 
     var modelURL: URL? {
         Bundle.main.url(forResource: Constants.modelName, withExtension: Constants.modelExtension)
     }
 
-    var storageURL: URL? {
-        userLibraryDirectory?.appendingPathComponent("\(Constants.modelName).\(Constants.storeExtension)")
+    var storageDirectory: URL {
+        switch storageLocation {
+        case .shared:
+            return sharedUserLibraryDirectory
+        case .legacy:
+            return legacyUserLibraryDirectory
+        }
+    }
+
+    var storageURL: URL {
+        switch storageLocation {
+        case .shared:
+            return sharedStorageURL
+        case .legacy:
+            return legacyStorageURL
+        }
+    }
+
+    var legacyBackupURL: URL {
+        legacyStorageURL.appendingPathExtension(Constants.oldExtension)
+    }
+
+    var legacyUserLibraryDirectory: URL {
+        let libraryURL = fileManager.urls(for: .libraryDirectory, in: .userDomainMask).last!
+        return libraryURL.appendingPathComponent(Constants.modelName)
+    }
+
+    var sharedUserLibraryDirectory: URL {
+        fileManager.sharedContainerURL.appendingPathComponent(Constants.dataDirectory)
+    }
+
+    var sharedStorageURL: URL {
+        sharedUserLibraryDirectory.appendingPathComponent("\(Constants.modelName).\(Constants.storeExtension)")
+    }
+
+    var legacyStorageURL: URL {
+        legacyUserLibraryDirectory.appendingPathComponent("\(Constants.modelName).\(Constants.storeExtension)")
+    }
+
+    func setStorageLocation(to newLocation: StorageLocation) {
+        storageLocation = newLocation
```

Comment:
I was thinking more in the lines of: having the logic, in here, to determine which one should be used (rather than having a parameter), in a dynamic property.

**Just** in case?

Reply:
done